### ISSUE TITLE
React-Navigation (instead of Navigator) Compatibility

### ISSUE THREAD

**Issue by nuttylord** [oblador/react-native-lightbox#74]
I am writing a project that navigates using react-navigation and i want to use lightbox however lightbox requires Navigator to function properly. I was hoping to see if i could write react-navigation support into the component and then allow a PR to happen. The possibility in my mind is using Navigator + react navigation so that i dont have to re-make my navigation because you could have both "this.props.navigator" and "this.props.navigation" but i am not sure how some of the navigation functions work.

I have been looking into lightbox.js at the occurences of Navigator and i am searching through the react-navigation api to see if i can find replacements for the functions used as it only occurs 8 times (2 of which are just checks to see if navigator is present).

lightbox.js - navigator occurences 
(cant get the formatting right) 

' 
             open: function() {
             this._root.measure((ox, oy, width, height, px, py) => {
             this.props.onOpen();

            this.setState({
              isOpen: (this.props.navigator ? true : false),
              isAnimating: true,
              origin: {
                width,
                height,
                x: px,
                y: py,
              },
            }, () => {
              if(this.props.navigator) {`

> create a route that will work with navigation (for react-navigation it would be the navigationOptions)

`                var route = {
                  component: LightboxOverlay,
                  passProps: this.getOverlayProps(),
                };`

> fetch the current set of routes

`                var routes = this.props.navigator.getCurrentRoutes();`

> push a route onto the stack

`                routes.push(route);`

> some sort of reset for the stack that Navigator does (still looking into it, any help knowing what this does would be nice.

`                      this.props.navigator.immediatelyResetRouteStack(routes);`

> Then the rest of the code for opening the lightbox runs

`                    } else {
                       this.setState({
                         isOpen: true,
                       });
                     }
                     this.setTimeout(() => {
                       this.state.layoutOpacity.setValue(0);
                     });
                   });
                 });
               },
`

for closing the stack the same lines are called but with .pop() instead of .push()

Anyone with experience of both react-navigation and react-native-lightbox would be a great help.

**Comment by acomito** [comment]
@nuttylord I'm about to undertake this. Did you ever find a solution?

**Comment by nuttylord** [comment]
no, i havent found a solution, but i hashed up a solution in notes that im going to attempt soon. basically my idea to try is to have a page to navigate to that just shows the image or whatever react-native to because what react-native-lightbox does is add a page to Navigators stack with .pop() and .push(). react-navigation doesnt seem to have this functionality (it might though! ive not tested it!) so what i was going to do was just add the `var route = { component: LightboxOverlay, passProps: this.getOverlayProps(), };` with react-navigations formatting for routes and just use .navigate("LightboxOverlay"). 

The possible problems i can think of without having tried anything is that react-navigation might not be able to have a component as a route or it might prevent the lightbox from applying its animation. Either way good luck!

**Comment by nuttylord** [comment]
@acomito Did you find a solution for this?

**Comment by acomito** [comment]
No @nuttylord I ended up avoiding a lightbox in the app for now.

**Comment by jayshah123** [comment]
Turns out Lightbox can work without navigator, in which case it uses Modal. 
But I had a weird case, where passing height or layout related props in style to Lightbox did not work. 
But worked when only gave only backgroundColor in style to LightBox component.

**Comment by acomito** [comment]
I'm getting some weird animation on exit. The animation in works great though.

**Comment by oblador** [comment]
I too want to make it work with react-navigation as it seems to become the standard navigator. I'm waiting on a PR to enable custom navigations more easily, but they've been unable to merge it for 6 months or so now. Not overly optimistic it will happen actually. I'm kind of reviving this repo and am happy to receive help if anyone is up for it as I'm a bit tight on free time.

**Comment by acomito** [comment]
I was able to get it working with react-navigation by just passing navigator={null}

My only issue was the exit animation was a little weird (images were in a sort of 3 column grid/rows via flexbox). But I think that was just flexbox not your package?

react-navigation was suppose to add "shared element transitions", but not sure where that's at. I would think that could be used to build a lightbox pretty easily.

https://github.com/react-community/react-navigation/issues/175
